api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,290 @@
+// should be set.
+type AdminNetworkPolicySubject struct {
+	NamespaceSelector       *metav1.LabelSelector `json:"namespaceselector"`
+	NamespaceAndPodSelector *NamespacedPodSubject `json:"namespaceandpodselector"`
+}
+
+// NamespacedPodSubject allows the user to select a given set of pod(s) in
+// selected namespace(s)
+type NamespacedPodSubject struct {
+	// This field follows standard label selector semantics; if present but empty,
+	// it selects all Namespaces.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceselector"`
+
+	// Used to explicitly select pods within a namespace; if present but empty,
+	// it selects all Pods.
+	PodSelector *metav1.LabelSelector `json:"podselector"`
+}
+
+// AdminNetworkPolicySpec defines the desired state of AdminNetworkPolicy
+type AdminNetworkPolicySpec struct {
+	// Priority is an int32 value bound to 0 - 1000, the lowest positive priority,
+	// "1" corresponds to the highest importance, while higher priorities have
+	// lower importance. An ANP with a priority of "0" will be evaluated after all
+	// positive priority AdminNetworkPolicies and standard NetworkPolicies.
+	// This field is NOT optional.
+	Priority *int32 `json:"priority"`
+
+	// Subject defines the objects to which this AdminNetworkPolicy applies.
+	// This field is NOT optional.
+	Subject AdminNetworkPolicySubject `json:"subject"`
+
+	// List of Ingress rules to be applied to the selected objects.
+	// A total of 100 rules will be allowed per each network policy instance,
+	// this rule count will be calculated as the total summation of the
+	// Ingress and Egress rules in a single AdminNetworkPolicy Instance.
+	// If this field is empty then this AdminNetworkPolicy has no effect on
+	// ingress traffic.
+	Ingress []AdminNetworkPolicyIngressRule `json:"ingress,omitempty"`
+
+	// List of Egress rules to be applied to the selected objects.
+	// A total of 100 rules will be allowed per each network policy instance,
+	// this rule count will be calculated as the total summation of the
+	// Ingress and Egress rules in a single AdminNetworkPolicy Instance.
+	// If this field is empty then this AdminNetworkPolicy has no effect on
+	// egress traffic.
+	Egress []AdminNetworkPolicyEgressRule `json:"egress,omitempty"`
+}
+
+// AdminNetworkPolicyIngressRule describes an action to take on a particular
+// set of traffic destined for pods selected by an AdminNetworkPolicy's
+// Subject field. The traffic must match both ports and from.
+type AdminNetworkPolicyIngressRule struct {
+	// Name is an identifier for this rule, that should be no more than 100 characters
+	// in length.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Action specifies whether this rule must pass, allow or deny traffic.
+	// Allow: allows the selected traffic
+	// Deny: denies the selected traffic
+	// Pass: allows the selected traffic to skip and remaining positive priority (non-zero)
+	// ANP rules and be delegated by K8's Network Policy.
```

Comment:
"K8's" can just be removed, since we're already within the context of Kubernetes. 

I would probably reword the "Pass" description so that it doesn't include the word "allow" and is more explicit w.r.t its behavior something like this (but maybe better):

 "Pass: skips any remaining positive priority AdminNetworkPolicy rules which apply to the pod and passes execution to any NetworkPolicies if there are any that select the pod, or zero-priority AdminNetworkPolicies."

Reply:
Agreed I'll take a stab at rephrasing this